astropy/io/fits/convenience.py

```diff
@@ -72,11 +72,12 @@
 from .fitsrec import FITS_rec
 from ...units import Unit
 from ...units.format.fits import UnitScaleError
+from ...units import Quantity
 from ...extern import six
 from ...extern.six import string_types
 from ...utils.exceptions import AstropyUserWarning
 from ...utils.decorators import deprecated_renamed_argument
-
+from ...table.column import BaseColumn
```

Comment:
If necessary (to avoid a circular import) you can move this import within the `if table.has_mixin_columns:` block.